tests/lib/prepare.sh

```diff
@@ -524,18 +524,47 @@ EOF
     rm -rf "$UNPACK_DIR"
 }
 
-
-uc20_build_initramfs_kernel_snap() {
+install_core_initrd_deps() {
     # carries ubuntu-core-initframfs
     add-apt-repository ppa:snappy-dev/image -y
     # TODO: install the linux-firmware as the current version of
     # ubuntu-core-initramfs does not depend on it, but nonetheless requires it
     # to build the initrd
     apt install ubuntu-core-initramfs linux-firmware -y
+}
 
+uc22_build_initramfs_kernel_snap() {
     local ORIG_SNAP="$1"
     local TARGET="$2"
 
+    install_core_initrd_deps
+
+    # extract the kernel snap, including extracting the initrd from the kernel.efi
+    kerneldir="$(mktemp --tmpdir -d kernel-workdirXXXXXXXXXX)"
+
+    unsquashfs -f -d "${kerneldir}" "$ORIG_SNAP"
+    (
+        cd "${kerneldir}"
+        config="$(echo config-*)"
+        kernelver="${config#config-}"
+        objcopy -O binary -j .linux kernel.efi kernel.img-"${kernelver}"
+        ubuntu-core-initramfs create-initrd --kerneldir modules/"${kernelver}" --kernelver "${kernelver}" --firmwaredir firmware --output ubuntu-core-initramfs.img
+        ubuntu-core-initramfs create-efi --initrd ubuntu-core-initramfs.img --kernel kernel.img --output kernel.efi --kernelver "${kernelver}"
```

Comment:
I don't think this is what we want, this will build a new EFI, I think we probably want to keep doing the objcpy etc as we do on 20 to _re-use_ the initrd from the kernel snap, only injecting/changing our snap-bootstrap

the changes we make in the 20 function to /usr/lib/the-tool are not necessary though, I expect some tests may be broken on UC22 but otherwise should boot

Reply:
Nice, thanks for the explanation, I'll try that